### ISSUE TITLE
"Get Jenkins admin password from file." fails

### ISSUE THREAD

**Issue by angel-manuel** [geerlingguy/ansible-role-jenkins#182]
While installing over docker image centos:7

```
    docker-centos: TASK [geerlingguy.jenkins : Get Jenkins admin password from file.] *************
    docker-centos: fatal: [default]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}
```

My playbook is

```yml
---

- hosts: all
  roles:
    - role: geerlingguy.java
      java_packages:
        - java-1.8.0-openjdk
    - role: geerlingguy.jenkins
      no_log: false #Also fails without this
      jenkins_package_state: latest
      jenkins_plugins_state: latest
      jenkins_http_port: 8080
      jenkins_admin_username: "admin"
      jenkins_admin_password_file: "../default-jenkins-password.txt"
      jenkins_plugins:
        - blueocean
        - git
        - pipeline-model-definition
      become: true
```

**Comment by ottosichert** [comment]
Same issue here. The password file task incorrectly handles `slurp` result in line 12:

https://github.com/geerlingguy/ansible-role-jenkins/blob/f5b11fbacfa17069163e29449e51a4a737aa4cd0/tasks/plugins.yml#L2-L13

I resolved it by manually retrieving the password before calling `geerlingguy.jenkins`:

```diff
 - name: get remote jenkins admin password file
   become: yes
   slurp:
     src: /var/lib/jenkins/secrets/initialAdminPassword
   register: adminpasswordfile
   no_log: True

 - name: set jenkins admin password
   set_fact:
+    jenkins_admin_password: "{{ adminpasswordfile.content }}"
-    jenkins_admin_password: "{{ adminpasswordfile.stdout }}"
   no_log: True

 - name: install jenkins
   become: yes
   include_role:
     name: geerlingguy.jenkins
```

**Comment by it-praktyk** [comment]
The default role configuration contains
[`jenkins_java_options: "-Djenkins.install.runSetupWizard=false"` ](https://github.com/geerlingguy/ansible-role-jenkins/blob/14f0e2155fdb0896878fcfd6b269d6dbc4406fed/defaults/main.yml#L17)
that cause that the file /var/lib/jenkins/secrets/initialAdminPassword is not created. I've found that information at [stackoverflow.com](https://stackoverflow.com/questions/35960883/how-to-unlock-jenkins); thak you @lukasz-gawron. 

@ottosichert, your solution is based on another mechanism used to create user and set password - [the groovy script](https://github.com/geerlingguy/ansible-role-jenkins/blob/14f0e2155fdb0896878fcfd6b269d6dbc4406fed/handlers/main.yml#L5). But I don't know what is the source of the mentioned file `/var/lib/jenkins/secrets/initialAdminPassword`. Previous installation of the Jenkins? A manual creation?

IMHO, all referenced to the `initialAdminPassword` are useless in the geerlingguy.jenkins role.

Generally, the geerlingguy.java role - in my opinion - is inconsistent and unmaintained so I've started creating my own base on that - you can find my fork https://github.com/it-praktyk/ansible-role-jenkins. Any constructive contribution will be welcomed.

**Comment by timblaktu** [comment]
I have run into this issue. My use of this role was working until today when I refactored a bit and changed some variable names. Not sure why this would cause this issue, but I did find that I am defining some variable names in an outer playbook-level group_var  which are the same name as the geerling role uses. This caused a [known recursion bug in ansible 2.9.10](https://github.com/ansible/ansible/issues/70279). 

@it-praktyk I may have a look at your fork and see if your removal of  initialAdminPassword helps me out.
